subprojects/testing-native/src/integTest/groovy/org/gradle/nativeplatform/test/xctest/SwiftXCTestIntegrationTest.groovy

```diff
@@ -82,21 +87,26 @@ apply plugin: 'swift-executable'
         succeeds("test")
 
         then:
-        result.assertTasksSkipped(":compileTestSwift", ":linkTest", ":createXcTestBundle", ":xcTest", ":test")
+        result.assertTasksSkipped(":compileSwift", ":compileTestSwift", ":linkTest", ":createXcTestBundle", ":xcTest", ":test")
     }
 
-    // TODO - Integrate automatically with the Swift executable
-    def "applying the xctest plugin together with swift-executable will not assemble the executable by convention"() {
+    def "xctest plugin can test public and internal feature of a Swift executable"() {
         def app = new SwiftApp()
         def test = new SwiftXcTestTestApp([
             newTestSuite("FooTestSuite", [
-                newTestCase("testFoo", TestElement.TestCase.Result.PASS)
-            ])
+                newTestCase("testInternalMultiply", TestElement.TestCase.Result.PASS,
+                    "XCTAssert(multiply(a: 21, b: 2) == ${app.multiply.multiply(21, 2)})"),
+                newTestCase("testPublicSum", TestElement.TestCase.Result.PASS,
+                    "XCTAssert(sum(a: 40, b: 2) == ${app.sum.sum(40, 2)})"),
+            ], ["App"])
         ])
 
         given:
+        settingsFile << "rootProject.name = 'App'"
         buildFile << """
 apply plugin: 'swift-executable'
+
+linkTest.source = project.files(new HashSet(linkTest.source.from)).filter { !it.name.equals("main.o") }
```

Comment:
We'll need to get rid of this, ideally in a way that allows me to test the classes defined in `main.swift` (but as another PR). I'm guessing this is excluded because we get a link error when linking the test executable?